source/jormungandr/jormungandr/scenarios/helper_classes/helper_utils.py

```diff
@@ -119,6 +124,118 @@ def _extend_pt_sections_with_fallback_sections(pt_journey, dp_journey):
+    car_park_section.origin.CopyFrom(car_park)
+    car_park_section.destination.CopyFrom(dp_extremity)
+    car_park_section.end_date_time = extremity_date_time
+    car_park_section.begin_date_time = car_park_section.end_date_time - CAR_PARK_DURATION
+    car_park_section.duration = CAR_PARK_DURATION
+
+    # crowfly_section
+    car_park_to_sp_section.type = response_pb2.SectionType.CROW_FLY
+    car_park_to_sp_section.street_network.CopyFrom(
+        response_pb2.StreetNetwork(duration=car_park_crowfly_duration, mode=response_pb2.Walking)
+    )
+    car_park_to_sp_section.origin.CopyFrom(pt_journey_extremity)
+    car_park_to_sp_section.destination.CopyFrom(car_park)
+    car_park_to_sp_section.end_date_time = car_park_section.begin_date_time
+    car_park_to_sp_section.begin_date_time = car_park_to_sp_section.end_date_time - car_park_crowfly_duration
+    car_park_to_sp_section.duration = car_park_crowfly_duration
+
+
+def _extend_with_car_park(
+    fallback_dp, pt_journey, fallback_type, walking_speed, car_park, car_park_crowfly_duration
+):
+    dp_journey = fallback_dp.journeys[0]
+
+    if fallback_type == StreetNetworkPathType.BEGINNING_FALLBACK:
+        dp_extremity = dp_journey.sections[-1].destination
+        # the first section of pt_journey is a fake crowfly section,
+        # the origin of the second section is the real start of the public transport journey
+        pt_journey_extrimity = pt_journey.sections[1].origin
+        car_park_section = dp_journey.sections.add()
+        car_park_to_sp_section = dp_journey.sections.add()
+
+        _make_beginning_car_park_sections(
+            car_park_section,
+            car_park_to_sp_section,
+            dp_extremity,
+            dp_journey.arrival_date_time,
+            pt_journey_extrimity,
+            car_park,
+            car_park_crowfly_duration,
+        )
+
+        dp_journey.arrival_date_time = car_park_to_sp_section.end_date_time
+
+    elif fallback_type == StreetNetworkPathType.ENDING_FALLBACK:
+
+        dp_extremity = dp_journey.sections[0].origin
+        # the last section of pt_journey is a fake crowfly section,
+        # the destination of the second section before the last is the real end of the public transport journey
+        pt_journey_extrimity = pt_journey.sections[-2].destination
+
+        car_park_section = dp_journey.sections.add()
+        car_park_to_sp_section = dp_journey.sections.add()
+
+        _make_ending_car_park_sections(
+            car_park_section,
+            car_park_to_sp_section,
+            dp_extremity,
+            dp_journey.departure_date_time,
+            pt_journey_extrimity,
+            car_park,
+            car_park_crowfly_duration,
+        )
+
+        dp_journey.departure_date_time = car_park_to_sp_section.begin_date_time
+
+    dp_journey.sections.sort(key=cmp_to_key(SectionSorter()))
+    dp_journey.duration += CAR_PARK_DURATION + car_park_crowfly_duration
```

Comment:
why we add _CAR_PARK_DURATION_ to duration ?

Reply:
because it's the whole thing (car + park + crowfly_to_stop_point) should be considered as a one direct_path